Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
+        }
+      }
+      catch (const MolSanitizeException &){
+        std::cerr << "Molecule could not be sanitized" << std::endl;
+      }
+    }
+  }
+  else{
+    mol=SmilesToMol(txt,0,sanitize,replacements);
+  }
+  return mol;
+}
+
+
+
+
 }  // end of namespace DaylightParserUtils
 
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
-    bool useSmiles) {
+    bool useSmiles, bool sanitize) {
```

Comment:
github code reviews are challenging places to have these kinds of discussions.... ah well.

I don't think that SMILES for reactions do need to always sanitize, it's easy to imagine cases where you want SMILES semantics (i.e. "C" just means "atomic number 6", not "atomic number 6 & aliphatic" like in SMARTS).

And I believe that (at least some forms of) matching work fine with unsantized queries. Here's an example:
```
In [8]: m = Chem.MolFromSmiles('c1cc([O-])cc(C)c1')

In [9]: p = Chem.MolFromSmiles('C:C(O):C',sanitize=False)

In [10]: m.HasSubstructMatch(p)
Out[10]: True

In [11]: Chem.MolFromSmiles('C1CC([O-])CC(C)C1').HasSubstructMatch(p)
Out[11]: False
```